### PR TITLE
implements articles loading bu routes static prefix and allow to load articles from route children's

### PR DESCRIPTION
**Pull request by ahilles107**
| Q                         | A
| ------------------------- | ---
| Bug fix?                  | yes
| New feature?              | yes
| BC breaks?                | no
| Deprecations?             | no
| Tests pass?               | yes
| Changelog update required | yes
| Fixed tickets             | SWP-1009
| License                   | AGPLv3

### CODE REVIEW

File: src/SWP/Bundle/ContentBundle/Doctrine/ORM/RouteRepository.php [hunk 1]
```diff
@@ -14,9 +14,21 @@
...
+use Doctrine\ORM\QueryBuilder;
 use SWP\Bundle\ContentBundle\Model\RouteRepositoryInterface;
 use SWP\Bundle\StorageBundle\Doctrine\ORM\EntityRepository;
 
 class RouteRepository extends EntityRepository implements RouteRepositoryInterface
 {
+    /**
+     * {@inheritdoc}
+     */
+    public function getChildrensByStaticPrefix(array $candidates, array $orderBy): QueryBuilder
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Doctrine/ORM/RouteRepository.php]: it should be `children` - `getChildrenByStaticPrefix`

File: src/SWP/Bundle/ContentBundle/Provider/RouteProviderInterface.php [hunk 2]
```diff
@@ -49,4 +49,27 @@ public function getOneById($id);
...
+     * @return array
+     */
+    public function getChildrensByStaticPrefix(array $candidates, array $orderBy = []): array;
+
+    /**
+     * @param array $candidates
+     *
+     * @return array|null
+     */
+    public function getWithChildrensByStaticPrefix(array $candidates): ?array;
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Provider/RouteProviderInterface.php]: same here: `children`

File: src/SWP/Bundle/ContentBundle/Provider/RouteProviderInterface.php [hunk 3]
```diff
@@ -49,4 +49,27 @@ public function getOneById($id);
...
+     */
+    public function getByStaticPrefix(array $candidates, array $orderBy = []): array;
+
+    /**
+     * @param array $candidates
+     * @param array $orderBy
+     *
+     * @return array
+     */
+    public function getChildrensByStaticPrefix(array $candidates, array $orderBy = []): array;
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Provider/RouteProviderInterface.php]: same here; `children`

File: src/SWP/Component/TemplatesSystem/Gimme/Context/Context.php [hunk 4]
```diff
@@ -371,4 +371,15 @@ public function restoreTemporaryUnset($id)
 
         return true;
     }
+
+    /**
+     *  Resets context data.
+     */
+    public function reset()
```
Comment (takeit) [inline src/SWP/Component/TemplatesSystem/Gimme/Context/Context.php]: is this method used somewhere besides the tests?

File: docs/manual/templates_system/articles.rst [hunk 5]
```diff
@@ -24,17 +24,28 @@ The :code:`articles` loader parameters:
...
 
+.. code-block:: twig
+
+    {% gimmelist article from articles with {'route': ['/news', '/sport/*']} %} <!-- route staticPrefix -->
+        <img src="{{ url(article) }}" />
+    {% endgimmelist %}
+
+.. note::
+
+   :code:`'/sport/*'` syntax will load articles from main route (:code:`'/sport'`) and all of it 1st level childrens (eg. :code:`'/sport/football'`).
```
Comment (takeit) [inline docs/manual/templates_system/articles.rst]: `children`

File: src/SWP/Bundle/ContentBundle/Doctrine/ORM/RouteRepository.php (hunk 1 above)
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Doctrine/ORM/RouteRepository.php]: `getChildrenByStaticPrefix`

File: src/SWP/Bundle/ContentBundle/Provider/ORM/RouteProvider.php [hunk 6]
```diff
@@ -112,6 +112,61 @@ public function getOneByStaticPrefix($staticPrefix)
...
+     */
+    public function getByStaticPrefix(array $candidates, array $orderBy = []): array
+    {
+        return $this->getRouteRepository()->findBy(['staticPrefix' => $candidates], $orderBy);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getChildrensByStaticPrefix(array $candidates, array $orderBy = []): array
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Provider/ORM/RouteProvider.php]: `getChildrenByStaticPrefix`

File: src/SWP/Bundle/ContentBundle/Provider/ORM/RouteProvider.php [hunk 7]
```diff
@@ -112,6 +112,61 @@ public function getOneByStaticPrefix($staticPrefix)
...
+    {
+        return $this->getRepository()->getChildrensByStaticPrefix($candidates, $orderBy)->getQuery()->getResult();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getWithChildrensByStaticPrefix(array $candidates): ?array
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Provider/ORM/RouteProvider.php]: `getWithChildrenByStaticPrefix`

File: src/SWP/Component/TemplatesSystem/Gimme/Context/Context.php (hunk 4 above)
Comment (ahilles107) [reply src/SWP/Component/TemplatesSystem/Gimme/Context/Context.php]: Not yet, but i couldn't find better way beside moving assertions to separate tests.

**Pull request merged**